tests/e2e/tests/e2e_util/util.py

```diff
@@ -57,11 +56,120 @@ def run_with_retry(
+            return p
+        except Exception as e:
+            if (
+                exceptions is None
+                or len(exceptions) == 0
+                or isinstance(e, exceptions)
+                or isinstance(e, AssertionError)
+            ):
+                LOG.info(f"Attempt {attempt}/{max_retries} failed. Error: {e}")
+                if attempt < max_retries:
+                    LOG.info(f"Retrying in {delay_between_retries} seconds...")
+                    time.sleep(delay_between_retries)
+                else:
+                    # If all attempts fail, raise the last error
+                    raise e
+
+
+def setup_network(h: harness.Harness, instance_id: str):
+    h.exec(instance_id, ["/snap/k8s/current/k8s/network-requirements.sh"])
+
+    LOG.info("Waiting for network to be enabled...")
+    retry_until_condition(
+        h,
+        instance_id,
+        ["k8s", "enable", "network"],
+        condition=lambda p: "enabled" in p.stderr.decode(),
+    )
+    LOG.info("Network enabled.")
+
+    LOG.info("Waiting for cilium pods to show up...")
+    retry_until_condition(
+        h,
+        instance_id,
+        [
+            "kubectl",
+            "--kubeconfig",
+            "/var/snap/k8s/common/etc/kubernetes/admin.conf",
+            "get",
+            "po",
+            "-n",
+            "kube-system",
+            "-o",
+            "json",
+        ],
+        condition=lambda p: "cilium" in p.stdout.decode(),
+    )
+    LOG.info("Cilium pods showed up.")
+
+    retry_until_condition(
+        h,
+        instance_id,
+        [
+            "kubectl",
+            "--kubeconfig",
+            "/var/snap/k8s/common/etc/kubernetes/admin.conf",
+            "wait",
+            "--for=condition=ready",
+            "pod",
+            "-n",
+            "kube-system",
+            "-l",
+            "io.cilium/app=operator",
+            "--timeout",
+            "180s",
+        ],
+        max_retries=3,
+        delay_between_retries=1,
+        check=True,
+    )
+
+    retry_until_condition(
+        h,
+        instance_id,
+        [
+            "kubectl",
+            "--kubeconfig",
+            "/var/snap/k8s/common/etc/kubernetes/admin.conf",
+            "wait",
+            "--for=condition=ready",
+            "pod",
+            "-n",
+            "kube-system",
+            "-l",
+            "k8s-app=cilium",
+            "--timeout",
+            "180s",
+        ],
+        max_retries=3,
+        delay_between_retries=1,
+        check=True,
+    )
+
+
 # Installs and setups the k8s snap on the given instance and connects the interfaces.
 def setup_k8s_snap(h: harness.Harness, instance_id: str, snap_path: Path):
     LOG.info("Install snap")
     h.send_file(instance_id, config.SNAP, snap_path)
     h.exec(instance_id, ["snap", "install", snap_path, "--dangerous"])
+    h.exec(instance_id, ["snap", "install", "kubectl", "--classic"])
```

Comment:
We still ship the kubectl binary at `/snap/k8s/current/bin/kubectl`.
No need to install it via snap.